godal.cpp

```diff
@@ -1725,3 +1725,40 @@ void godalGridCreate(cctx *ctx, char *pszAlgorithm, GDALGridAlgorithm eAlgorithm
 
 	godalUnwrap();
 }
+
+GDALDatasetH godalGrid(cctx *ctx, const char *pszDest, GDALDatasetH hSrcDS, char **switches) {
+	godalWrap(ctx);
+
+	GDALGridOptions *gridopts = GDALGridOptionsNew(switches,nullptr);
+	if(failed(ctx)) {
+		GDALGridOptionsFree(gridopts);
+		godalUnwrap();
+		return nullptr;
+	}
+
+	// Perform checks for gridding algorithm compatibility with this GDAL build here
+	char *algString;
+	int argc = CSLCount(switches);
+    for (int i = 0; i < argc && switches != nullptr && switches[i] != nullptr; i++) {
+		if (strcmp(switches[i], "-a") && (i+1) < argc) {
+			algString = switches[i+1];
+			break;
+		}
+	}
+	// "linear" gridding algorithm only compatible with GDAL builds with QHull support
+	if (!GDALHasTriangulation() && strcmp(algString, "linear") == 0) {
+		CPLError(CE_Failure, CPLE_AppDefined, "unable to run GGA_Linear algorithm, since GDAL built without QHull support");
+		godalUnwrap();
+		return nullptr;
+	}
```

Comment:
I'm not sure we want to do too much validation of available options inside godal, and leave that to GDAL itself. The following call to `GDALGrid` will return an error in case `linear` is selected and qhull is not supported right? If so that is sufficient from godal's point of view

Reply:
Yes that's correct, I'll commit your suggestion to remove it now